src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +38,32 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        splitAddress(address);
+        this.value = this.toString();
+    }
+
+    private void splitAddress(String address) {
+        ArrayList<String> detailedAddress = new ArrayList<>(Arrays.asList(address.split(",")));
+        if (detailedAddress.size() > BLOCK_INDEX) {
```

Comment:
Instead of making the values optional and checking for each value, maybe you could make it such that an address has to have all 4 values?